op-batcher/batcher/driver.go

```diff
@@ -388,17 +388,18 @@ func (l *BatchSubmitter) mainLoop(ctx context.Context, receiptsCh chan txmgr.TxR
 
 			l.state.pruneSafeBlocks(syncStatus.SafeL2)
 			l.state.pruneChannels(syncStatus.SafeL2)
+
+			err = l.checkExpectedProgress(*syncStatus)
```

Comment:
Yes agreed about adding a test I will get on to that.

I don't think the pruning should affect the progress check. If a channel got pruned, that is because all of the blocks in that channel became safe. So none of those pruned channels should trigger the "did not make progress" error condition (since `syncStatus.SafeL2.Number < ch.LatestL2().Number` would always be false).

Reply:
Ah I see, that makes sense.